### PR TITLE
util/testkit: refine testkit

### PR DESCRIPTION
**Pull request by coocood**
Now we don't need to convert a string to a byte slice to check results.
Also supports Sort, to make unordered test result stable.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @hanfei1991 @tiancaiamao PTAL

File: executor/executor_test.go [hunk 1]
```diff
@@ -305,23 +298,19 @@ func (s *testSuite) TestSelectOrderBy(c *C) {
 	tk.MustExec("begin")
 	// Test limit
 	r = tk.MustQuery("select * from select_order_test order by name, id limit 1 offset 0;")
-	rowStr = fmt.Sprintf("%v %v", 1, []byte("hello"))
-	r.Check(testkit.Rows(rowStr))
+	r.Check(testkit.Rows("1 hello"))
 	tk.MustExec("commit")
 
 	tk.MustExec("begin")
```
Comment (hanfei1991) [inline executor/executor_test.go]: remove the useless begin end

File: util/testkit/testkit.go [hunk 2]
```diff
@@ -34,21 +35,47 @@ type TestKit struct {
...
+	return ifacesSlice
+}
+
+// Sort sorts and return the result.
+func (res *Result) Sort() *Result {
+	sort.Slice(res.rows, func(i, j int) bool {
+		a := res.rows[i]
+		b := res.rows[j]
+		for i := range a {
+			if a[i] == "<nil>" {
```
Comment (hanfei1991) [inline util/testkit/testkit.go]: Do we need process it specially? It's ok to sort by alphabet order.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**